### PR TITLE
[incubator/fluent-cloudwatch] Fix fatal bug and Support EC2 instance role 

### LINKED ISSUE #4732: [incubator/fluentd-cloudwatch] header field value cannot include CR/LF
**Is this a request for help?**: Yes

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): BUG REPORT

**Version of Helm and Kubernetes**:
```
Client: &version.Version{SemVer:"v2.8.2", GitCommit:"a80231648a1473929271764b920a8e346f6de844", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.8.1", GitCommit:"6af75a8fd72e2aa18a2b278cfe5c7a1c5feca7f2", GitTreeState:"clean"}
```

```
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.0", GitCommit:"fc32d2f3698e36b93322a3465f63a14e9f0eaead", GitTreeState:"clean", BuildDate:"2018-03-29T10:10:42Z", GoVersion:"go1.9.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.4", GitCommit:"9befc2b8928a9426501d3bf62f72849d5cbcd5a3", GitTreeState:"clean", BuildDate:"2017-11-20T05:17:43Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Which chart**: incubator/fluentd-cloudwatch

**What happened**: I get an error when running the chart

```
2018-04-05 10:18:09 +0000 [warn]: temporarily failed to flush the buffer. next_retry=2018-04-05 10:18:10 +0000 error_class="ArgumentError" error="header field value cannot include CR/LF" plugin_id="object:2ac2e86fe1c4"
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/2.3.0/net/http/header.rb:20:in `block in initialize_http_header'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/2.3.0/net/http/header.rb:16:in `each'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/2.3.0/net/http/header.rb:16:in `initialize_http_header'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/2.3.0/net/http/generic_request.rb:44:in `initialize'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/2.3.0/net/http/request.rb:15:in `initialize'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-3.19.0/lib/seahorse/client/net_http/handler.rb:143:in `new'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-3.19.0/lib/seahorse/client/net_http/handler.rb:143:in `build_net_request'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-3.19.0/lib/seahorse/client/net_http/handler.rb:74:in `block in transmit'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-3.19.0/lib/seahorse/client/net_http/handler.rb:123:in `block in session'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-3.19.0/lib/seahorse/client/net_http/connection_pool.rb:99:in `session_for'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-3.19.0/lib/seahorse/client/net_http/handler.rb:121:in `session'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-3.19.0/lib/seahorse/client/net_http/handler.rb:73:in `transmit'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-3.19.0/lib/seahorse/client/net_http/handler.rb:47:in `call'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-3.19.0/lib/seahorse/client/plugins/content_length.rb:12:in `call'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-3.19.0/lib/aws-sdk-core/json/error_handler.rb:8:in `call'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-3.19.0/lib/aws-sdk-core/plugins/signature_v4.rb:66:in `call'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-3.19.0/lib/aws-sdk-core/plugins/helpful_socket_errors.rb:10:in `call'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-3.19.0/lib/aws-sdk-core/plugins/retry_errors.rb:92:in `call'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-3.19.0/lib/aws-sdk-core/json/handler.rb:11:in `call'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-3.19.0/lib/aws-sdk-core/plugins/user_agent.rb:13:in `call'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-3.19.0/lib/seahorse/client/plugins/endpoint.rb:45:in `call'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-3.19.0/lib/aws-sdk-core/plugins/param_validator.rb:24:in `call'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-3.19.0/lib/seahorse/client/plugins/raise_response_errors.rb:14:in `call'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-3.19.0/lib/aws-sdk-core/plugins/jsonvalue_converter.rb:20:in `call'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-3.19.0/lib/aws-sdk-core/plugins/idempotency_token.rb:17:in `call'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-3.19.0/lib/aws-sdk-core/plugins/param_converter.rb:24:in `call'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-3.19.0/lib/aws-sdk-core/plugins/response_paging.rb:10:in `call'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-3.19.0/lib/seahorse/client/plugins/response_target.rb:23:in `call'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-3.19.0/lib/seahorse/client/request.rb:70:in `send_request'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-cloudwatchlogs-1.2.0/lib/aws-sdk-cloudwatchlogs/client.rb:725:in `describe_log_groups'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.4.5/lib/fluent/plugin/out_cloudwatch_logs.rb:403:in `log_group_exists?'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.4.5/lib/fluent/plugin/out_cloudwatch_logs.rb:152:in `block in write'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.4.5/lib/fluent/plugin/out_cloudwatch_logs.rb:144:in `each'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.4.5/lib/fluent/plugin/out_cloudwatch_logs.rb:144:in `write'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluentd-0.12.33/lib/fluent/buffer.rb:354:in `write_chunk'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluentd-0.12.33/lib/fluent/buffer.rb:333:in `pop'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluentd-0.12.33/lib/fluent/output.rb:342:in `try_flush'
  2018-04-05 10:18:09 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluentd-0.12.33/lib/fluent/output.rb:149:in `run'
```

**What you expected to happen**: No errors

**How to reproduce it** (as minimally and precisely as possible): Run with minimal config and defaults

```
rbac:
  create: true
awsRegion: '<REGION>'
logGroupName: '<LOG GROUP>'
```

**Anything else we need to know**: The error seems to be coming from the cloudwatch fluentd plugin. I have raised ryotarai/fluent-plugin-cloudwatch-logs#101 with them.

### LINKED ISSUE #5810: [incubator/fluentd-cloudwatch] AWS authorization errors
**Is this a request for help?**: Yes

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): BUG REPORT

**Version of Helm and Kubernetes**:
```
Client: &version.Version{SemVer:"v2.8.2", GitCommit:"a80231648a1473929271764b920a8e346f6de844", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.8.2", GitCommit:"a80231648a1473929271764b920a8e346f6de844", GitTreeState:"clean"}
```
Kubernetes 1.10

**Which chart**: incubator/fluentd-cloudwatch

**What happened**: Fluentd does not seem to be able to authorize with AWS.

With my AWS access key and secret key added in the `secrets` directory:
```
$ helm install --set awsRegion=eu-west-1 --set rbac.create=true --set image.tag=v0.12.33-cloudwatch@sha256:0a6763c174ac9456ae3b71ae4485ff5f9ab7ecd5a1542e71248a72c54666f02c ./workspace/charts/incubator/fluentd-cloudwatch

2018-05-29 14:52:14 +0000 [warn]: temporarily failed to flush the buffer. next_retry=2018-05-29 14:52:15 +0000 error_class="Aws::CloudWatchLogs::Errors::IncompleteSignatureException" error="Authorization header requires 'Signature' parameter. Authorization header requires 'SignedHeaders' parameter. Authorization=AWS4-HMAC-SHA256 Credential=[redacted]" plugin_id="object:2b10a5734ca0"
  2018-05-29 14:52:14 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.10.50/lib/seahorse/client/plugins/raise_response_errors.rb:15:in `call'
  2018-05-29 14:52:14 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.10.50/lib/aws-sdk-core/plugins/jsonvalue_converter.rb:20:in `call'
  2018-05-29 14:52:14 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.10.50/lib/aws-sdk-core/plugins/idempotency_token.rb:18:in `call'
  2018-05-29 14:52:14 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.10.50/lib/aws-sdk-core/plugins/param_converter.rb:20:in `call'
  2018-05-29 14:52:14 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.10.50/lib/aws-sdk-core/plugins/response_paging.rb:26:in `call'
  2018-05-29 14:52:14 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.10.50/lib/seahorse/client/plugins/response_target.rb:21:in `call'
  2018-05-29 14:52:14 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.10.50/lib/seahorse/client/request.rb:70:in `send_request'
  2018-05-29 14:52:14 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.10.50/lib/seahorse/client/base.rb:207:in `block (2 levels) in define_operation_methods'
  2018-05-29 14:52:14 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.4.0/lib/fluent/plugin/out_cloudwatch_logs.rb:298:in `log_group_exists?'
  2018-05-29 14:52:14 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.4.0/lib/fluent/plugin/out_cloudwatch_logs.rb:121:in `block in write'
  2018-05-29 14:52:14 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.4.0/lib/fluent/plugin/out_cloudwatch_logs.rb:113:in `each'
  2018-05-29 14:52:14 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.4.0/lib/fluent/plugin/out_cloudwatch_logs.rb:113:in `write'
  2018-05-29 14:52:14 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluentd-0.12.33/lib/fluent/buffer.rb:354:in `write_chunk'
  2018-05-29 14:52:14 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluentd-0.12.33/lib/fluent/buffer.rb:333:in `pop'
  2018-05-29 14:52:14 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluentd-0.12.33/lib/fluent/output.rb:342:in `try_flush'
  2018-05-29 14:52:14 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluentd-0.12.33/lib/fluent/output.rb:149:in `run'
```

With `kube2iam` installed and using the `awsRole` flag:

```
$ helm install --set awsRegion=eu-west-1 --set rbac.create=true --set image.tag=v0.12.33-cloudwatch@sha256:0a6763c174ac9456ae3b71ae4485ff5f9ab7ecd5a1542e71248a72c54666f02c --set awsRole=ecs_rw incubator/fluentd-cloudwatch

2018-05-29 14:09:25 +0000 [warn]: temporarily failed to flush the buffer. next_retry=2018-05-29 14:09:26 +0000 error_class="Aws::Errors::MissingCredentialsError" error="unable to sign request without credentials set" plugin_id="object:2ab1696c42fc"
  2018-05-29 14:09:25 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.10.50/lib/aws-sdk-core/plugins/request_signer.rb:104:in `require_credentials'
  2018-05-29 14:09:25 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.10.50/lib/aws-sdk-core/plugins/request_signer.rb:94:in `sign_authenticated_requests'
  2018-05-29 14:09:25 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.10.50/lib/aws-sdk-core/plugins/request_signer.rb:87:in `call'
  2018-05-29 14:09:25 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.10.50/lib/aws-sdk-core/plugins/helpful_socket_errors.rb:10:in `call'
  2018-05-29 14:09:25 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.10.50/lib/aws-sdk-core/plugins/retry_errors.rb:89:in `call'
  2018-05-29 14:09:25 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.10.50/lib/aws-sdk-core/json/handler.rb:11:in `call'
  2018-05-29 14:09:25 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.10.50/lib/aws-sdk-core/plugins/user_agent.rb:12:in `call'
  2018-05-29 14:09:25 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.10.50/lib/seahorse/client/plugins/endpoint.rb:41:in `call'
  2018-05-29 14:09:25 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.10.50/lib/aws-sdk-core/plugins/param_validator.rb:21:in `call'
  2018-05-29 14:09:25 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.10.50/lib/seahorse/client/plugins/raise_response_errors.rb:14:in `call'
  2018-05-29 14:09:25 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.10.50/lib/aws-sdk-core/plugins/jsonvalue_converter.rb:20:in `call'
  2018-05-29 14:09:25 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.10.50/lib/aws-sdk-core/plugins/idempotency_token.rb:18:in `call'
  2018-05-29 14:09:25 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.10.50/lib/aws-sdk-core/plugins/param_converter.rb:20:in `call'
  2018-05-29 14:09:25 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.10.50/lib/aws-sdk-core/plugins/response_paging.rb:26:in `call'
  2018-05-29 14:09:25 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.10.50/lib/seahorse/client/plugins/response_target.rb:21:in `call'
  2018-05-29 14:09:25 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.10.50/lib/seahorse/client/request.rb:70:in `send_request'
  2018-05-29 14:09:25 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.10.50/lib/seahorse/client/base.rb:207:in `block (2 levels) in define_operation_methods'
  2018-05-29 14:09:25 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.4.0/lib/fluent/plugin/out_cloudwatch_logs.rb:298:in `log_group_exists?'
  2018-05-29 14:09:25 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.4.0/lib/fluent/plugin/out_cloudwatch_logs.rb:121:in `block in write'
  2018-05-29 14:09:25 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.4.0/lib/fluent/plugin/out_cloudwatch_logs.rb:113:in `each'
  2018-05-29 14:09:25 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluent-plugin-cloudwatch-logs-0.4.0/lib/fluent/plugin/out_cloudwatch_logs.rb:113:in `write'
  2018-05-29 14:09:25 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluentd-0.12.33/lib/fluent/buffer.rb:354:in `write_chunk'
  2018-05-29 14:09:25 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluentd-0.12.33/lib/fluent/buffer.rb:333:in `pop'
  2018-05-29 14:09:25 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluentd-0.12.33/lib/fluent/output.rb:342:in `try_flush'
  2018-05-29 14:09:25 +0000 [warn]: /usr/lib/ruby/gems/2.3.0/gems/fluentd-0.12.33/lib/fluent/output.rb:149:in `run'
```

**What you expected to happen**: Fluentd should be able to connect correctly.

**How to reproduce it** (as minimally and precisely as possible): See commands used above.

**Anything else we need to know**: My AWS user account has permissions to view and create Cloudwatch log groups. The `ecs_rw` role has permissions like so:

```
{
 "Version": "2012-10-17",
 "Statement": [
   {
     "Effect": "Allow",
     "Action": [
       "logs:*"
     ],
     "Resource": "*"
   }
 ]
}
```

I am specifying the image tag as a workaround recommended [here](https://github.com/kubernetes/charts/issues/5196).

If this a permissions problem, is it possible to document the required permissions here? I am happy to do a PR if helpful.

### PR DESCRIPTION
**Pull request by masahiro331**
**What this PR does / why we need it**:
1. I want to use EC2 instance iam role. Now,  when arnRole and awsCredentials are empty, using "<AWS_ACCESS_KEY_ID>", "<AWS_SECRET_ACCESS_KEY>". 
2. This pull request fixes default authorization header issue that empty credential,
 "<AWS_ACCESS_KEY_ID>\n", is always used without customized credential setting. It enable to use EC2 instance role credential by default. 

This is error log. 
```
2018-09-13 01:51:26 +0000 [warn]: temporarily failed to flush the buffer. next_retry=2018-09-13 01:51:27 +0000 error_class="ArgumentError" error="header field value cannot include CR/LF" plugin_id="object:2ab5bbf17b10"
  2018-09-13 01:51:26 +0000 [warn]: /usr/lib/ruby/2.4.0/net/http/header.rb:25:in `block in initialize_http_header'
  2018-09-13 01:51:26 +0000 [warn]: /usr/lib/ruby/2.4.0/net/http/header.rb:16:in `each'
  2018-09-13 01:51:26 +0000 [warn]: /usr/lib/ruby/2.4.0/net/http/header.rb:16:in `initialize_http_header'
  2018-09-13 01:51:26 +0000 [warn]: /usr/lib/ruby/2.4.0/net/http/generic_request.rb:44:in `initialize'
  2018-09-13 01:51:26 +0000 [warn]: /usr/lib/ruby/2.4.0/net/http/request.rb:15:in `initialize'
  2018-09-13 01:51:26 +0000 [warn]: /fluentd/vendor/bundle/ruby/2.4.0/gems/aws-sdk-core-3.22.1/lib/seahorse/client/net_http/handler.rb:146:in `new'
  2018-09-13 01:51:26 +0000 [warn]: /fluentd/vendor/bundle/ruby/2.4.0/gems/aws-sdk-core-3.22.1/lib/seahorse/client/net_http/handler.rb:146:in `build_net_request'
  2018-09-13 01:51:26 +0000 [warn]: /fluentd/vendor/bundle/ruby/2.4.0/gems/aws-sdk-core-3.22.1/lib/seahorse/client/net_http/handler.rb:74:in `block in transmit'
  2018-09-13 01:51:26 +0000 [warn]: /fluentd/vendor/bundle/ruby/2.4.0/gems/aws-sdk-core-3.22.1/lib/seahorse/client/net_http/handler.rb:126:in `block in session'
  2018-09-13 01:51:26 +0000 [warn]: /fluentd/vendor/bundle/ruby/2.4.0/gems/aws-sdk-core-3.22.1/lib/seahorse/client/net_http/connection_pool.rb:100:in `session_for'
  2018-09-13 01:51:26 +0000 [warn]: /fluentd/vendor/bundle/ruby/2.4.0/gems/aws-sdk-core-3.22.1/lib/seahorse/client/net_http/handler.rb:121:in `session'
  2018-09-13 01:51:26 +0000 [warn]: /fluentd/vendor/bundle/ruby/2.4.0/gems/aws-sdk-core-3.22.1/lib/seahorse/client/net_http/handler.rb:73:in `transmit'
  2018-09-13 01:51:26 +0000 [warn]: /fluentd/vendor/bundle/ruby/2.4.0/gems/aws-sdk-core-3.22.1/lib/seahorse/client/net_http/handler.rb:47:in `call'
  2018-09-13 01:51:26 +0000 [warn]: /fluentd/vendor/bundle/ruby/2.4.0/gems/aws-sdk-core-3.22.1/lib/seahorse/client/plugins/content_length.rb:12:in `call'
  2018-09-13 01:51:26 +0000 [warn]: /fluentd/vendor/bundle/ruby/2.4.0/gems/aws-sdk-core-3.22.1/lib/aws-sdk-core/json/error_handler.rb:8:in `call'
  2018-09-13 01:51:26 +0000 [warn]: /fluentd/vendor/bundle/ruby/2.4.0/gems/aws-sdk-core-3.22.1/lib/aws-sdk-core/plugins/signature_v4.rb:66:in `call'
  2018-09-13 01:51:26 +0000 [warn]: /fluentd/vendor/bundle/ruby/2.4.0/gems/aws-sdk-core-3.22.1/lib/aws-sdk-core/plugins/helpful_socket_errors.rb:10:in `call'
  2018-09-13 01:51:26 +0000 [warn]: /fluentd/vendor/bundle/ruby/2.4.0/gems/aws-sdk-core-3.22.1/lib/aws-sdk-core/plugins/retry_errors.rb:138:in `call'
  2018-09-13 01:51:26 +0000 [warn]: /fluentd/vendor/bundle/ruby/2.4.0/gems/aws-sdk-core-3.22.1/lib/aws-sdk-core/json/handler.rb:11:in `call'
  2018-09-13 01:51:26 +0000 [warn]: /fluentd/vendor/bundle/ruby/2.4.0/gems/aws-sdk-core-3.22.1/lib/aws-sdk-core/plugins/user_agent.rb:13:in `call'
  2018-09-13 01:51:26 +0000 [warn]: /fluentd/vendor/bundle/ruby/2.4.0/gems/aws-sdk-core-3.22.1/lib/seahorse/client/plugins/endpoint.rb:45:in `call'
  2018-09-13 01:51:26 +0000 [warn]: /fluentd/vendor/bundle/ruby/2.4.0/gems/aws-sdk-core-3.22.1/lib/aws-sdk-core/plugins/param_validator.rb:24:in `call'
  2018-09-13 01:51:26 +0000 [warn]: /fluentd/vendor/bundle/ruby/2.4.0/gems/aws-sdk-core-3.22.1/lib/seahorse/client/plugins/raise_response_errors.rb:14:in `call'
  2018-09-13 01:51:26 +0000 [warn]: /fluentd/vendor/bundle/ruby/2.4.0/gems/aws-sdk-core-3.22.1/lib/aws-sdk-core/plugins/jsonvalue_converter.rb:20:in `call'
  2018-09-13 01:51:26 +0000 [warn]: /fluentd/vendor/bundle/ruby/2.4.0/gems/aws-sdk-core-3.22.1/lib/aws-sdk-core/plugins/idempotency_token.rb:17:in `call'
  2018-09-13 01:51:26 +0000 [warn]: /fluentd/vendor/bundle/ruby/2.4.0/gems/aws-sdk-core-3.22.1/lib/aws-sdk-core/plugins/param_converter.rb:24:in `call'
  2018-09-13 01:51:26 +0000 [warn]: /fluentd/vendor/bundle/ruby/2.4.0/gems/aws-sdk-core-3.22.1/lib/aws-sdk-core/plugins/response_paging.rb:10:in `call'
  2018-09-13 01:51:26 +0000 [warn]: /fluentd/vendor/bundle/ruby/2.4.0/gems/aws-sdk-core-3.22.1/lib/seahorse/client/plugins/response_target.rb:23:in `call'
  2018-09-13 01:51:26 +0000 [warn]: /fluentd/vendor/bundle/ruby/2.4.0/gems/aws-sdk-core-3.22.1/lib/seahorse/client/request.rb:70:in `send_request'
  2018-09-13 01:51:26 +0000 [warn]: /fluentd/vendor/bundle/ruby/2.4.0/gems/aws-sdk-cloudwatchlogs-1.4.0/lib/aws-sdk-cloudwatchlogs/client.rb:736:in `describe_log_groups'
  2018-09-13 01:51:26 +0000 [warn]: /fluentd/vendor/bundle/ruby/2.4.0/gems/fluent-plugin-cloudwatch-logs-0.4.5/lib/fluent/plugin/out_cloudwatch_logs.rb:403:in `log_group_exists?'
  2018-09-13 01:51:26 +0000 [warn]: /fluentd/vendor/bundle/ruby/2.4.0/gems/fluent-plugin-cloudwatch-logs-0.4.5/lib/fluent/plugin/out_cloudwatch_logs.rb:152:in `block in write'
  2018-09-13 01:51:26 +0000 [warn]: /fluentd/vendor/bundle/ruby/2.4.0/gems/fluent-plugin-cloudwatch-logs-0.4.5/lib/fluent/plugin/out_cloudwatch_logs.rb:144:in `each'
  2018-09-13 01:51:26 +0000 [warn]: /fluentd/vendor/bundle/ruby/2.4.0/gems/fluent-plugin-cloudwatch-logs-0.4.5/lib/fluent/plugin/out_cloudwatch_logs.rb:144:in `write'
  2018-09-13 01:51:26 +0000 [warn]: /fluentd/vendor/bundle/ruby/2.4.0/gems/fluentd-0.12.43/lib/fluent/buffer.rb:354:in `write_chunk'
  2018-09-13 01:51:26 +0000 [warn]: /fluentd/vendor/bundle/ruby/2.4.0/gems/fluentd-0.12.43/lib/fluent/buffer.rb:333:in `pop'
  2018-09-13 01:51:26 +0000 [warn]: /fluentd/vendor/bundle/ruby/2.4.0/gems/fluentd-0.12.43/lib/fluent/output.rb:342:in `try_flush'
  2018-09-13 01:51:26 +0000 [warn]: /fluentd/vendor/bundle/ruby/2.4.0/gems/fluentd-0.12.43/lib/fluent/output.rb:149:in `run'
```

dump /usr/lib/ruby/2.4.0/net/http/header.rb:16 "key: #{key}, value: #{value}" 
```
"key: authorization, value: AWS4-HMAC-SHA256 Credential=<AWS_ACCESS_KEY_ID>\n/20180913/ap-northeast-1/logs/aws4_request, SignedHeaders=content-type;host;x-amz-content-sha256;x-amz-date;x-amz-target, Signature=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
```
env["AWS_ACCESS_KEY_ID"] == "<AWS_ACCESS_KEY_ID>\n"

**Which issue this PR fixes** : 
This is related to the issue helm/helm#3410

fixes #5810
fixes #4732

**Special notes for your reviewer**:
@jmcarp
@icereval 
@kenden 

thank you!!

### CODE REVIEW

**Comment by masahiro331** [comment]
Fix README

**Comment by masahiro331** [comment]
Thank you cpanato!!!!!
I'm Happy

File: incubator/fluentd-cloudwatch/README.md [hunk 1]
```diff
@@ -22,8 +22,10 @@ This chart bootstraps a [Fluentd](https://www.fluentd.org/) [Cloudwatch](https:/
 To install the chart with the release name `my-release`:
 
 ```console
-$ # edit secrets/aws_access_key_id and secrets/aws_access_key_id with the key/password of a AWS user with a policy to access  Cloudwatch
+$ # use ec2 instance role credential
```
Comment (cpanato) [inline incubator/fluentd-cloudwatch/README.md:25]: and if I don't want to use the ec2 role?

Comment (masahiro331) [reply incubator/fluentd-cloudwatch/README.md:25]: use other credential 

```
$ # or add aws_access_key_id and aws_access_key_id with the key/password of a AWS user with a policy to access  Cloudwatch
$ helm install --name my-release incubator/fluentd-cloudwatch --set awsAccessKeyId=aws_access_key_id_here --set awsSecretAccessKey=aws_secret_access_key_here
$ # or add a role to aws with the correct policy to add to cloud watch
$ helm install --name my-release incubator/fluentd-cloudwatch --set awsRole=roll_name_here
```

Comment (cpanato) [reply incubator/fluentd-cloudwatch/README.md:25]: ah got it

**Comment by cpanato** [comment]
testing locally...

**Comment by cpanato** [comment]
works locally

/lgtm

**Comment by masahiro331** [comment]
please assign approver

**Pull request merged**